classes/DbWrapper.php

```diff
@@ -0,0 +1,149 @@
+     * @param array<mixed> $data
+     *
+     * @throws UpdateDatabaseException
+     */
+    public static function update(string $table, array $data, string $where = '', int $limit = 0, bool $null_values = false, bool $use_cache = true, bool $add_prefix = true): bool
+    {
+        $result = Db::getInstance()->update($table, $data, $where, $limit, $null_values, $use_cache, $add_prefix);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     */
+    public static function execute(string $sql, bool $use_cache = true): bool
+    {
+        $result = Db::getInstance()->execute($sql, $use_cache);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     *
+     * @return array<mixed>|bool|mysqli_result|PDOStatement|resource|null
+     */
+    public static function executeS(string $sql, bool $array = true, bool $use_cache = true)
+    {
+        $result = Db::getInstance()->executeS($sql, $array, $use_cache);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     *
+     * @return string|false|null Returns false if no results
+     */
+    public static function getValue(string $sql, bool $use_cache = true)
+    {
+        $result = Db::getInstance()->getValue($sql, $use_cache);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @param array<mixed> $data
+     *
+     * @throws UpdateDatabaseException
+     */
+    public static function insert(string $table, array $data, bool $null_values = false, bool $use_cache = true, int $type = Db::INSERT, bool $add_prefix = true): bool
+    {
+        $result = Db::getInstance()->insert($table, $data, $null_values, $use_cache, $type, $add_prefix);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @return int|string
+     *
+     * @throws UpdateDatabaseException
+     */
+    public static function Insert_ID()
+    {
+        $result = Db::getInstance()->Insert_ID();
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     */
+    public static function delete(string $table, string $where = '', int $limit = 0, bool $use_cache = true, bool $add_prefix = true): bool
+    {
+        $result = Db::getInstance()->delete($table, $where, $limit, $use_cache, $add_prefix);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     *
+     * @return array<mixed>|bool|object|null
+     */
+    public static function getRow(string $sql, bool $use_cache = true)
+    {
+        $result = Db::getInstance()->getRow($sql, $use_cache);
+        self::validateDBQuery();
+
+        return $result;
+    }
+
+    /**
+     * @throws UpdateDatabaseException
+     */
+    public static function validateDBQuery(): void
```

Comment:
What about:

```suggestion
    public static function assertQuerySuccess(): void
```

Reply:
I don't expect to have an exception on an assertion, but rather a boolean? This would require having to manage exceptions on all calls